modules/doc/content/newsletter/2019_07.md

```diff
@@ -42,3 +73,50 @@ Two simple generic constant materials were also added that can be used in array
 
 - [GenericConstantArray.md]
 - [GenericConstant2DArray.md]
+
+## Robust Race-condition checking in the TestHarness
+
+One of the more common errors in regression test writing is the accidental creation of output file race-conditions
+when multiple conditions are tested with the same input file or multiple input files writing to the same output file.
+While the TestHarness has long had the ability to check for common race conditions, we found that users were
+able to sneak in unexpected race conditions quite frequently. Our MOOSE intern (Samuel Tew) has recently completed
+a "robust race-condition" checker that is now checked in and is being used on all PRs to MOOSE. The new system works
+by isolating each test in the same directory and checking the file-system +before+ and +after+ each test runs
+examining all file system activity. This makes it possible to look at the tests that may run concurrently and cross-examine
+them with the outputs created by those tests when they run separately looking for conflicts. You will be notified
+by CIVET when a potential race-condition is detected. You may also run this mode locally by using the `-d` option
+to the `run_tests` script.
+
+## TimedPrint
+
+A new utility has been added to MOOSE that will conditionally print progress information in the form of timed
+dots on the terminal when specific blocks of code take an "extended" period of time. At the completion of the
+timed block of code, an elapsed time is also printed giving the user information about which activity is consuming
+non-trivial amounts of time. No information is printed if the section of code executes quickly. Developers may
+use this capability in blocks of code that are expected to take significant amounts of time. Note: This capability
+should not be used in frequently called sections of code.
+
+```
+#include "TimedPrint.h"
+
+...
+
+{
+  CONSOLE_TIMED_PRINT("Running a time-consuming algorithm");
+
+
+
+} // Dots are printed until the closing scope is reached as time elapses
+```
+
+## Improved Build Parallelism on GNU Make 4.0+
+
+Makefile rules have been improved so that additionally parallelism is possible when using systems with newer
+GNU Make installed. Users on many flavors of Linux may notice significant reductions in build times when building
+applications that use several libraries or modules in MOOSE.
+
+## SQA Push
+
+The MOOSE team is making a big push to get all tests documented in the moose/test folder. This activity started
+in June and will continue through July. As part of this effort, we have extended/improved the syntax available
+for documenting tests. If you are working on SQA, please see the new syntax available [here](https://mooseframework.inl.gov/python/MooseDocs/generate.html).
```

Comment:
```suggestion
for documenting tests. If you are working on SQA, please see the new syntax available in [MooseDocs/generate.md].
```

Reply:
@permcody This is how you should reference the content.

Reply:
The content still hasn't been updated yet. Do you have a PR coming for that?

Reply:
I made a suggestion for you to look at, is it not there?